Without the build (`dist` folder), `dioxus serve` fails

Hi everyone!

Without the `dist` folder, `dioxus serve` fails with `I/O Error: No such file or directory (os error 2)`.
Using dioxus CLI ver. `0.1.5`.

```shell
❯ dioxus serve
[ERROR] 🚫 Serve startup failed: I/O Error: No such file or directory (os error 2)
❯ 
```

This happens right after creating a new project (as the build is not done automatically) or if you remove it (or use `dioxus clean`).

A simple message such as _"You need to build it before starting it."_ would improve the user (dev) experience.

--- a/src/cli/serve/mod.rs
+++ b/src/cli/serve/mod.rs
@@ -1,5 +1,6 @@
 use super::*;
 use std::{
+    fs::create_dir_all,
     io::Write,
     path::PathBuf,
     process::{Command, Stdio},
@@ -71,19 +72,19 @@ impl Serve {
     pub fn regen_dev_page(crate_config: &CrateConfig) -> Result<()> {
         let serve_html = gen_page(&crate_config.dioxus_config, true);
 
-        let mut file = std::fs::File::create(
+        let dist_path = crate_config.crate_dir.join(
             crate_config
-                .crate_dir
-                .join(
-                    crate_config
-                        .dioxus_config
-                        .application
-                        .out_dir
-                        .clone()
-                        .unwrap_or_else(|| PathBuf::from("dist")),
-                )
-                .join("index.html"),
-        )?;
+                .dioxus_config
+                .application
+                .out_dir
+                .clone()
+                .unwrap_or_else(|| PathBuf::from("dist")),
+        );
+        if !dist_path.is_dir() {
+            create_dir_all(&dist_path)?;
+        }
+        let index_path = dist_path.join("index.html");
+        let mut file = std::fs::File::create(index_path)?;
         file.write_all(serve_html.as_bytes())?;
 
         Ok(())

